### PR TITLE
Projectiles now orbit blackholes

### PR DESCRIPTION
**Pull request by hal9000PR**
## What Does This PR Do
Projectiles will now curve around singulos. 
I planned on doing more for this but this took me ages. hardest PR to lines written ratio ever.
If I get more time I will do more reworking so its actually an interesting engine.

## Why It's Good For The Game
neat feature

## Images of changes

dreamseeker_zbUYinyVa1

## Testing
you dont want to fucking know what I have been through.

## Changelog
:cl:
experiment: projectiles now curve around black holes. 
/:cl:

### CODE REVIEW

**Comment by procdrone** [comment]
to clarify, what happens to emitter shots that power containment? will they also orbit, miss the shield gens, and make the singularity engine loose every time?

File: code/modules/power/singularity/singularity.dm [hunk 1]
```diff
@@ -444,3 +446,25 @@
 	eat()
 	if(prob(1))
 		mezzer()
+
+
+/obj/singularity/HasProximity(atom/movable/AM)
+	if(istype(AM, /obj/item/projectile))
```
Comment (S34NW) [inline code/modules/power/singularity/singularity.dm:452]: Exclude `/obj/item/projectile/beam/emitter` from this

**Review by S34NW — CHANGES_REQUESTED** [review]

File: code/modules/power/singularity/singularity.dm (hunk 1 above)
Comment (S34NW) [reply code/modules/power/singularity/singularity.dm]: Note: This is so that you can actually have a functional containment for singulos

**Comment by S34NW** [comment]
Any update on this? Currently sat in limbo for a little while.

**Comment by hal9000PR** [comment]
> to clarify, what happens to emitter shots that power containment? will they also orbit, miss the shield gens, and make the singularity engine loose every time?

[ezgif.com-gif-maker.webm](https://user-images.githubusercontent.com/69320440/216192011-0ac0676a-7f84-4d9f-99e8-3da74aef180e.webm)

**Review by S34NW — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**